lib/Model/AbstractModel.php

```diff
@@ -282,6 +283,14 @@ protected static function getModelFactory()
         return \Pimcore::getContainer()->get('pimcore.model.factory');
     }
 
+    /**
+     * @return ClassResolver
+     */
+    protected static function getDocumentClassResolver(): ClassResolver
```

Comment:
@brusch 
Should we use a this for every type or should rather use the container call directly where we need it, or like add a param, which resolver to get?

Reply:
I'd go on the container directly